### PR TITLE
Added support for json input from command line in example

### PR DESCRIPTION
**Pull request by p-gentili**
Given #145 and the TODO comment inside the example main-gen.c I think this PR may be useful for those who need to encode more complicated json.

### CODE REVIEW

File: examples/main-gen.c [hunk 1]
```diff
@@ -82,9 +86,15 @@ int main(int argc, char *argv[])
 				}
 			}
 			break;
+		case 'j':
+			free(opt_json);
+			opt_json = (char *) malloc((strlen(optarg)+1)*sizeof(char));
+			strcpy(opt_json, optarg);
```
Comment (benmcollins) [inline examples/main-gen.c:92]: Let's use `strdup()` instead, and please check for NULL.

File: examples/main-gen.c [hunk 2]
```diff
@@ -55,6 +56,9 @@ int main(int argc, char *argv[])
 		char *val;
 	} opt_claims[100];
 	memset(opt_claims, 0, sizeof(opt_claims));
+	char* opt_json;
+	opt_json = (char *) malloc(3*sizeof(char));
+	strcpy(opt_json, "{}");
```
Comment (benmcollins) [inline examples/main-gen.c:61]: Set this to `opt_json = NULL` so we only create this when it's passed on the command line.

File: examples/main-gen.c [hunk 3]
```diff
@@ -116,6 +126,8 @@ int main(int argc, char *argv[])
 		jwt_add_grant(jwt, opt_claims[i].key, opt_claims[i].val);
 	}
 
+	ret = jwt_add_grants_json(jwt, opt_json);
```
Comment (benmcollins) [inline examples/main-gen.c:129]: if (opt_json != NULL) {
        ret = jwt_add_grants_json(jwt, opt_json);
        /* Add check for error in ret */
    }

File: examples/main-gen.c [hunk 4]
```diff
@@ -131,6 +143,7 @@ int main(int argc, char *argv[])
 
 	jwt_free_str(out);
 finish:
+	free(opt_json);
```
Comment (benmcollins) [inline examples/main-gen.c:146]: if (opt_json != NULL)
        free(opt_json);

**Review by benmcollins — CHANGES_REQUESTED** [review]
Change the title to show "from command line" since this is not `stdin`

Thanks for the changes.

**Comment by p-gentili** [comment]
Just updated with your suggestions. Thanks.

**Pull request merged**